docs/API.md

```diff
@@ -310,25 +310,26 @@ tracer.use('pg', {
+| -------------------- | ------------------------------ | ----------- | ----------- |
+| enabled              | `DD_TRACE_ENABLED`             | `true`      | Whether to enable the tracer. |
+| debug                | `DD_TRACE_DEBUG`               | `false`     | Enable debug logging in the tracer. |
+| service              | `DD_SERVICE_NAME`              | -           | The service name to be used for this program. |
+| url                  | `DD_TRACE_AGENT_URL`           | -           | The url of the trace agent that the tracer will submit to. Takes priority over hostname and port, if set. |
+| hostname             | `DD_TRACE_AGENT_HOSTNAME`      | `localhost` | The address of the agent that the tracer will submit to. |
+| port                 | `DD_TRACE_AGENT_PORT`          | `8126`      | The port of the trace agent that the tracer will submit to. |
+| dogstatsd.port       | `DD_DOGSTATSD_PORT`            | `8125`      | The port of the Dogstatsd agent that metrics will be submitted to. |
+| env                  | `DD_ENV`                       | -           | Set an application’s environment e.g. `prod`, `pre-prod`, `stage`. |
+| logInjection         | `DD_LOGS_INJECTION`            | `false`     | Enable automatic injection of trace IDs in logs for supported logging libraries. |
+| tags                 | -                              | `{}`        | Set global tags that should be applied to all spans. |
+| sampleRate           | -                              | `1`         | Percentage of spans to sample as a float between 0 and 1. |
+| flushInterval        | -                              | `2000`      | Interval in milliseconds at which the tracer will submit traces to the agent. |
+| runtimeMetrics       | `DD_RUNTIME_METRICS_ENABLED`   | `false`     | Whether to enable capturing runtime metrics. Port 8125 (or configured with `dogstatsd.port`) must be opened on the agent for UDP. |
+| reportHostname       | `DD_TRACE_REPORT_HOSTNAME`     | `false`     | Whether to report the system's hostname for each trace. When disabled, the hostname of the agent will be used instead. |
+| experimental         | -                              | `{}`        | Experimental features can be enabled all at once using boolean `true` or individually using key/value pairs. Please contact us to learn more about the available experimental features. |
+| plugins              | -                              | `true`      | Whether or not to enable automatic instrumentation of external libraries using the built-in plugins. |
+| clientToken          | `DD_CLIENT_TOKEN`              | -           | Client token for browser tracing. Can be generated in the UI at `Integrations -> APIs`. |
+| integrationsDisabled | `DD_INTEGRATIONS_DISABLED`     | `[]`        | Set an array of integration names to disable upon tracer is initialization, e.g. `['express','http','dns']` or `'express,http,dns'` when set as env var. |
```

Comment:
Let's reuse the `plugins` options instead. Maybe it could support an object in addition to a boolean, with a structure like this:

```js
plugins: {
  express: false // DD_TRACE_DISABLED_PLUGINS=express
}
```

We want to eventually move away from `tracer.use()` and rely only on the option, and this would be more in line with that goal.

Reply:
I took at shot at this, the `plugin` config option /`DD_TRACE_DISABLED_PLUGINS`env var is a little bit unwieldy now with this implementation given the various forms it can take (`true/false/'true'/'false'/{<object_with_key_value_pairs>}/'comma,seperated,string'`) depending on how it's set. I added a helper function to `config.js` to handle coalescing these various formats into one value, tried to write it as clearly as possible but let me know if this is just becoming too confusing a use case and should be separated into it's own option. It was a bit hard to document properly as well.

Reply:
I think only a list of plugin names is enough. Disabling all plugins can lead to many issues and we want to deprecate that in the future, so we should not make it easier to do that.

Configuring plugins in the env var is also not needed since it many cases options take a function as their value which can only be done in code.

Reply:
@rochdev gotcha i think that makes sense to me, so to be clear you're saying just something like `plugins: ['express', 'http']` or `DD_TRACE_DISABLED_PLUGINS='express,http'` would be appropriate here, with just the config (_non env var_) option maintaining the current support for a boolean (_though it's planned to be deprecated in future versions_). can update accordingly

Reply:
@rochdev updated so that the configuration option passed in code can be either a boolean (`true/false`) or an array of integrations to disabled, and passed as environment variable can only be a comma separated string of integration names to disable. I think this also helps simplify a lot of the `config.js` code needed for coalescing the plugins value. lmk what you think.